impress-org/givewp
# Refactor: update models repositories to mutate model instances

## Description



When count was added to the model factories during https://github.com/impress-org/givewp/commit/ecd9878275174ada39397f9a5513ae170a06d15c the created instances of model factories were not being returned - only their initial instantiated instances.  This resulted in model factories not returning important properties like their ID and created_at which caused all the model tests to fail.

This brought us to discuss how model repositories _should_ handle the given instance.  The conclusion was to let the repositories mutate the model instance and not return anything.

This means the api is changing slightly, so instead of returning a new instance every time, the current instance will be updated.

Here is a basic example

```
$model = new Model();
$model->foo = 'bar';
$model->save()
$model->id // 1
$model->foo // 'bar'
``` 

During this PR a few other updates were made:

1. The Subscription model was missing an important conditional in its save method.
2. Change action `give_subscription_deleted` to `give_subscription_model_deleted`. There was already an action with this name so needed to avoid conflict.
3. added `afterMaking` method to model factory.
4. Updated default model factory properties that require ids with factory creations (relationships)
5. Updated all Model `create` methods to return the model instance
6. Updated all Model `save` methods to return void
7. Updated all Model `insert` methods to return void
8. Updated all model repositories to update their instances on insert.
9. Updated all model & repository tests to reflect the change.

## Affects



- Donor, Donation, and Subscription models & how they use create & save
- Donor, Donation, and Subscription repositories & how they use insert & update
- All Models and Model Repository tests

## Testing Instructions



1. Make sure when running Model::factory()->create() it actually returns the created instance - not the instantiated instance.
2. Play around with the Model create & save methods.
3. Run the model & repository unit tests. 

## Pre-review Checklist
